src/picongpu/CMakeLists.txt

```diff
@@ -254,6 +254,25 @@ if( (Boost_VERSION EQUAL 105500) AND
       "${CUDA_NVCC_FLAGS} \"-DBOOST_NOINLINE=__attribute__((noinline))\" ")
 endif()
 
+# PMacc (ab)uses boost::result_of for non-functors. This define forces boost to use the
+# result<> member template of the target type
+if(Boost_VERSION LESS 105500)
+    add_definitions(-DBOOST_RESULT_OF_USE_TR1)
+else()
+    # Boost 1.55 adds support for another define that makes result_of look for the result<> template
+    # and falls back to decltype if none is found. This is great for the transition from the "wrong"
+    # usage to the "correct" one as both can be used. But:
+    # 1) Cannot be used before 7.5 due to nvcc bug:
+    #    http://stackoverflow.com/questions/31940457/make-nvcc-output-traces-on-compile-error
```

Comment:
hm, not a very helpful link regarding that.

with nvcc 6.5, boost 1.54.0 and gcc 4.8.2 I can compile the example with `nvcc -std=c++11 ...` without any flags.

Disabling `-std=c++11` does neither work with that version of gcc nor nvcc.

are you sure you posted the right link? It does not even mention `BOOST_RESULT_OF_USE_TR1*` flags o.0